src/manifolds/ProductManifold.jl

```diff
@@ -129,21 +129,21 @@ end
     check_manifold_point(M::ProductManifold, p; kwargs...)
 
 Check whether `p` is a valid point on the [`ProductManifold`](@ref) `M`.
+If `p` is not a point on `M` a `CompositeException` consisting of all error messages of the
+components, for which the tests fail is returned.
 
 The tolerance for the last test can be set using the `kwargs...`.
 """
-function check_manifold_point(M::ProductManifold, p::ProductRepr; kwargs...)
-    for t in ziptuples(M.manifolds, submanifold_components(M, p))
-        err = check_manifold_point(t...; kwargs...)
-        err === nothing || return err
-    end
-    return nothing
-end
-
-function check_manifold_point(M::ProductManifold, p::ProductArray; kwargs...)
-    for t in ziptuples(M.manifolds, submanifold_components(M, p))
-        err = check_manifold_point(t...; kwargs...)
-        err === nothing || return err
+function check_manifold_point(
+    M::ProductManifold,
+    p::Union{ProductRepr,ProductArray};
+    kwargs...,
+)
+    ts = ziptuples(M.manifolds, submanifold_components(M, p))
+    e = [check_manifold_point(t...; kwargs...) for t in ts]
+    errors = filter((x) -> !isnothing(x), e)
+    if length(errors) > 0
```

Comment:
You can also do `count(x -> !isnothing(x), e)` instead of filtering and calling `length`.

Reply:
But what is I also only want to return the array of errors? That's mainly what I am filtering for.

Reply:
Yes, you are right, but this made me realize -- will there be sufficient information in the error to tell which components the errors correspond to? I get the filtering now but maybe we should zip `e` with component indices before filtering?

Reply:
I am not 100% sure it would work (with CompositeException) but it might, and would be helpful, yes.

Reply:
I thought a little bit about this, I don't feel so god about collecting something else then errors in the CompositeException, though the type it accepts seems to alsow for that. Do you think it might be enough that our error messages usually repeat the data that is wrong?

Reply:
So yes it does work, but the composite error prints quite strange then (so if it does not contain an array of errors).

Reply:
Finally I checked whether one can “edit” an error message, and no, they are immutable; since we do not know which errors appear, w can not that easily prepend something like `"On Comp #1:` before the error.

Reply:
Right, that printing is a problem. Ideally I guess we would have a separate exception type that encapsulates errors for specific components and stores component index, and has custom printing. Anyway, what you already did here is definitely an improvement over the current behavior so we can ignore these indices.